### PR TITLE
fix: avoid error converting user requested stop for custom searcher

### PR DESCRIPTION
**Pull request by mpkouznetsov**
## Description

When using custom searcher, user requested stop (from code) generates a panic.

```
unexpected exited reason: 0xc00189c720
goroutine 892 [running]: runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65 github.com/determined-ai/determined/master/pkg/actor.(*Ref).close(0xc00138c6c0)
	/home/ubuntu/code/determined/master/pkg/actor/ref.go:309 +0x22d panic({0x2a31660, 0xc00189c730})
	/usr/local/go/src/runtime/panic.go:838 +0x207 github.com/determined-ai/determined/master/pkg/model.(*ExitedReason).ToProto(0xb409fe2fc20fb6a?)
	/home/ubuntu/code/determined/master/pkg/model/experiment.go:870 +0xee github.com/determined-ai/determined/master/pkg/searcher.(*customSearch).trialExitedEarly(0xc001177fb0, {0xc0018a5080?, 0xc00062d0f8?}, {0x6a, 0xfb, 0x20, 0xfc, 0xe2, 0x9f, 0x40, ...}, ...)
	/home/ubuntu/code/determined/master/pkg/searcher/custom_search.go:109 +0x105 github.com/determined-ai/determined/master/pkg/searcher.(*Searcher).TrialExitedEarly(0xc00101da00, {0x6a, 0xfb, 0x20, 0xfc, 0xe2, 0x9f, 0x40, 0xb, 0x99, ...}, ...)
	/home/ubuntu/code/determined/master/pkg/searcher/searcher.go:115 +0x26a github.com/determined-ai/determined/master/internal.(*experiment).Receive(0xc0005e9380, 0xc001917a00)
	/home/ubuntu/code/determined/master/internal/experiment.go:295 +0x1a45 github.com/determined-ai/determined/master/pkg/actor.(*Ref).processMessage(0xc00138c6c0)
	/home/ubuntu/code/determined/master/pkg/actor/ref.go:256 +0x632 github.com/determined-ai/determined/master/pkg/actor.(*Ref).run(0xc00138c6c0)
	/home/ubuntu/code/determined/master/pkg/actor/ref.go:268 +0xb5 created by github.com/determined-ai/determined/master/pkg/actor.newRef
	/home/ubuntu/code/determined/master/pkg/actor/ref.go:94 +0x576   actor-local-addr="33" actor-system="master" experiment-id="33" go-type="experiment" job-id="e40aee8b-3c3e-4de1-b6ed-9d7c0772250d"

```
## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by stoksc — APPROVED** [review]

File: master/pkg/model/experiment.go [hunk 1]
```diff
@@ -857,15 +857,17 @@ func ExitedReasonFromProto(r trialv1.TrialEarlyExit_ExitedReason) ExitedReason {
 	}
 }
 
-// ToProto converts an ExitedReason to its protobuf representation.
-func (r *ExitedReason) ToProto() experimentv1.TrialExitedEarly_ExitedReason {
+// ToSearcherProto converts an ExitedReason to its protobuf representation for searcher purposes.
+func (r *ExitedReason) ToSearcherProto() experimentv1.TrialExitedEarly_ExitedReason {
 	switch *r {
```
Comment (stoksc) [inline master/pkg/model/experiment.go:862]: not sure why this has a pointer receive, just to deref it

File: master/pkg/model/experiment.go [hunk 2]
```diff
@@ -857,15 +857,17 @@ func ExitedReasonFromProto(r trialv1.TrialEarlyExit_ExitedReason) ExitedReason {
...
 	case Errored:
 		return *experimentv1.TrialExitedEarly_EXITED_REASON_UNSPECIFIED.Enum()
```
Comment (stoksc) [inline master/pkg/model/experiment.go:864]: similar comment here, just return `experimentv1.TrialExitedEarly_EXITED_REASON_UNSPECIFIED`, looks like `.Enum()` is just fancy syntax to get a pointer

**Review by nrajanee — APPROVED** [review]
lgtm 
+1 to bradley's comments

File: master/pkg/model/experiment.go (hunk 1 above)
Comment (mpkouznetsov) [reply master/pkg/model/experiment.go]: A case of cargo-culting by a golang amateur :)

**Pull request merged**